### PR TITLE
Removed pch.h from shared items projects

### PR DESCRIPTION
**Pull request by vmoroz**
Some of our shared items projects had pch.h files. When such shared project is added to another project with pch.h we cannot rely on which file is being used for PCH.
In this change:
- Removed pch.h from Chakra and Shared items projects
- Make sure that ReactWindowsCore uses pch.h by forcing pch.h include and defined its usage in project file.
- Some minor changes to source code to eliminate compiler warnings.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3958)

### CODE REVIEW

File: change/react-native-windows-2020-01-24-13-55-58-MS_NoSharedItemsPCH.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "type": "prerelease",
+  "comment": "Removed pch.h from shared items projects",
+  "packageName": "react-native-windows",
+  "email": "vmorozov@microsoft.com",
```
Comment (NikoAri) [inline change/react-native-windows-2020-01-24-13-55-58-MS_NoSharedItemsPCH.json]: >vmorozov [](start = 12, length = 8)

Should this be jshost?

File: vnext/Chakra/ChakraExecutor.cpp [hunk 2]
```diff
@@ -250,7 +250,7 @@ void ChakraExecutor::initOnJSVMThread() {
   SystraceSection s("ChakraExecutor.initOnJSVMThread");
 
   bool enableDebugging = m_instanceArgs.EnableDebugging;
-  bool breakOnNextLine = m_instanceArgs.DebuggerBreakOnNextLine;
+  [[maybe_unused]] bool breakOnNextLine = m_instanceArgs.DebuggerBreakOnNextLine;
```
Comment (NikoAri) [inline vnext/Chakra/ChakraExecutor.cpp]: >[[maybe_unused]] [](start = 2, length = 16)

why not use if #DEBUG macro?

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj [hunk 3]
```diff
@@ -78,6 +78,7 @@
   <Import Project="..\Chakra\Chakra.vcxitems" Label="Shared" />
   <Import Project="..\JSI\Shared\JSI.Shared.vcxitems" Label="Shared" />
   <Import Project="..\Shared\Shared.vcxitems" Label="Shared" />
+  <Import Project="..\Mso\Mso.vcxitems" Label="Shared" />
```
Comment (NikoAri) [inline vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj]: >Mso [](start = 22, length = 3)

(nit) would be nice if these were ordered :)

File: vnext/ReactUWP/Modules/NativeUIManager.cpp [hunk 4]
```diff
@@ -31,7 +31,7 @@ static YogaNodePtr make_yoga_node() {
 static int YogaLog(const YGConfigRef config, const YGNodeRef node, YGLogLevel level, const char *format, va_list args) {
   int len = _scprintf(format, args);
   std::string buffer(len + 1, '\0');
-  int c = vsnprintf_s(&buffer[0], len + 1, _TRUNCATE, format, args);
+  [[maybe_unused]] int c = vsnprintf_s(&buffer[0], len + 1, _TRUNCATE, format, args);
```
Comment (NikoAri) [inline vnext/ReactUWP/Modules/NativeUIManager.cpp]: >c [](start = 23, length = 1)

why not remove it?

File: vnext/ReactUWP/Modules/Animated/FacadeType.h [hunk 5]
```diff
@@ -5,7 +5,7 @@
 #include <cassert>
 #include <string>
 
-enum FacadeType {
+enum class FacadeType {
```
Comment (NikoAri) [inline vnext/ReactUWP/Modules/Animated/FacadeType.h]: >class [](start = 5, length = 5)

Awesome!!!

File: vnext/ReactUWP/Modules/Animated/FacadeType.h [hunk 6]
```diff
@@ -19,7 +19,7 @@ enum FacadeType {
   Progress
 };
 
-static FacadeType StringToFacadeType(const std::string &string) {
+inline FacadeType StringToFacadeType(const std::string &string) {
```
Comment (NikoAri) [inline vnext/ReactUWP/Modules/Animated/FacadeType.h]: >string [](start = 56, length = 6)

it's a bit confusing to use variables same as language primitives. And although C++ "string" is not a true primitive, it's still confusing.

File: vnext/ReactUWP/Modules/Animated/NativeAnimatedNodeManager.cpp [hunk 7]
```diff
@@ -164,7 +164,7 @@ void NativeAnimatedNodeManager::StartTrackingAnimatedNode(
           updatedAnimationConfig.insert(
               static_cast<folly::StringPiece>(s_framesName), [animationConfig, activeFrames = item.second->Frames()]() {
                 auto frames = folly::dynamic::array();
-                for (auto const &frame : animationConfig.find("frames").dereference().second) {
+                for ([[maybe_unused]] auto const &frame : animationConfig.find("frames").dereference().second) {
```
Comment (NikoAri) [inline vnext/ReactUWP/Modules/Animated/NativeAnimatedNodeManager.cpp]: >&frame [](start = 49, length = 6)

Can't we just do a simple for (unsigned int i=0; i<animationConfig.find("frames").dereference().second.length; i++){frames.push_back(0.0);}
?

File: vnext/ReactUWP/Threading/WorkerMessageQueueThread.cpp [hunk 8]
```diff
@@ -88,7 +88,7 @@ struct WorkerMessageQueueThread::Impl {
 };
 
 WorkerMessageQueueThread::Impl::Impl() {
-  HRESULT hr = CreateAsyncWorkQueue(&queue);
+  [[maybe_unused]] HRESULT hr = CreateAsyncWorkQueue(&queue);
```
Comment (NikoAri) [inline vnext/ReactUWP/Threading/WorkerMessageQueueThread.cpp]: >HRESULT hr [](start = 19, length = 10)

Remove or comment out if it's not used?

File: change/react-native-windows-2020-01-24-13-55-58-MS_NoSharedItemsPCH.json (hunk 1 above)
Comment (vmoroz) [reply change/react-native-windows-2020-01-24-13-55-58-MS_NoSharedItemsPCH.json]: No, it is an e-mail of a person who does changes.

---
In reply to: [370869851](https://github.com/microsoft/react-native-windows/pull/3958#discussion_r370869851) [](ancestors = 370869851)

File: vnext/Chakra/ChakraExecutor.cpp (hunk 2 above)
Comment (vmoroz) [reply vnext/Chakra/ChakraExecutor.cpp]: This is a standard way to say that variable maybe unused. It is used below under USE_EDGEMODE_JSRT condition. The #DEBUG will not do a right thing here.

---
In reply to: [370870108](https://github.com/microsoft/react-native-windows/pull/3958#discussion_r370870108) [](ancestors = 370870108)

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj (hunk 3 above)
Comment (vmoroz) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj]: VS will re-order them after sometime anyway. It does not seem to be worth the effort...

---
In reply to: [370870305](https://github.com/microsoft/react-native-windows/pull/3958#discussion_r370870305) [](ancestors = 370870305)

File: vnext/ReactUWP/Modules/Animated/FacadeType.h (hunk 6 above)
Comment (vmoroz) [reply vnext/ReactUWP/Modules/Animated/FacadeType.h]: The main issue here was the 'static' keyword that forces compiler create a unique code of the function. You have a good point - I will change it.

---
In reply to: [370870951](https://github.com/microsoft/react-native-windows/pull/3958#discussion_r370870951) [](ancestors = 370870951)

File: vnext/ReactUWP/Modules/Animated/NativeAnimatedNodeManager.cpp (hunk 7 above)
Comment (vmoroz) [reply vnext/ReactUWP/Modules/Animated/NativeAnimatedNodeManager.cpp]: We probably could, but I do not know the underlying implementation details and how to test that change.

---
In reply to: [370871461](https://github.com/microsoft/react-native-windows/pull/3958#discussion_r370871461) [](ancestors = 370871461)

File: vnext/ReactUWP/Threading/WorkerMessageQueueThread.cpp (hunk 8 above)
Comment (vmoroz) [reply vnext/ReactUWP/Threading/WorkerMessageQueueThread.cpp]: commented out

---
In reply to: [370871722](https://github.com/microsoft/react-native-windows/pull/3958#discussion_r370871722) [](ancestors = 370871722)

File: vnext/ReactUWP/Modules/NativeUIManager.cpp (hunk 4 above)
Comment (vmoroz) [reply vnext/ReactUWP/Modules/NativeUIManager.cpp]: removed

---
In reply to: [370870518](https://github.com/microsoft/react-native-windows/pull/3958#discussion_r370870518) [](ancestors = 370870518)

**Pull request merged**